collector/cloudwatch/classes/collector.php

```diff
@@ -0,0 +1,88 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+namespace cltr_cloudwatch;
+
+use local_aws\local\client_factory;
+use tool_cloudmetrics\collector\base;
+use tool_cloudmetrics\metric\metric_item;
+
+defined('MOODLE_INTERNAL') || die();
+
+require_once($CFG->dirroot . '/local/aws/sdk/aws-autoloader.php');
+
+
+/**
+ * Collector class for AWS Cloudwatch.
+ *
+ * @package   cltr_cloudwatch
+ * @author    Jason den Dulk <jasondendulk@catalyst-au.net>
+ * @copyright 2022, Catalyst IT
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+class collector extends base {
+    protected static $client = null;
+    protected static $pluginconfig;
+
+    public function __construct() {
+        global $CFG;
+
+        if (is_null(self::$client)) {
+            $clientconfig = [
+                'region' => get_config('cltr_cloudwatch', 'awsregion'),
+                'version' => lib::AWS_VERSION,
+            ];
+
+            // Add AWS credentials if specified in CFG. IAM role and environment would apply if they are not set.
+            if (isset($CFG->forced_plugin_settings['cltr_cloudwatch']['credentials'])) {
```

Comment:
forced_plugin_settings is how you force set it, but for reads this comes through as normal get_config. If you read it from forced_plugin_settings then it will only be able to be set via forced_plugin_settings and it will ignore if set in the ui

Reply:
done